immich-app/immich
# feat(web): timeline bucket for albums (4)

Fixes #3606, #3607

Related to:
- #3550 
- #2370
- #2337

In this PR:
- Use time buckets on the album (not shared links though)
- Add link icon under album title
- Add owner icon to list of shared user icons
- Add owner item to list of users (share info modal)
- Add option to get album by id without return assets
- Include new property (`hasSharedLink`) to know if there is at least one shared link on the album
- Add typing to several dispatcher events
- Simplify `CreateSharedLinkModal` interface
- Add another `UserAvatar` size in between the previous `md` and `lg`.

Tested Scenarios:
- Create a new album from select in the timeline (no share button)
- Create a new album from asset view in the timeline
- Create a new shared album from sharing page (shows share button)
- Update album cover (thumbnail)
- Add user to album
- Remove user from album
- Remove self from album (leave album)
- Remove asset
- Remove assets via shift+select
- Remove assets via group select
- Add assets via shift-select
- Add assets via group select
- Add/remove assets and album details (date, count) are correctly updated
- Create album shared link
- View link icon in share row (if owner)
- No view link in share row (for non-owner)
- Asset view next/previous work as expected
- Delete from album shows up if all selected assets are owned by me (non owner)
- Delete from album is not visible if selection includes a non-owned asset
- Add asset from computer (owned)
- Add asset from computer (non owned) immediately adds a broken thumbnail, which requires a refresh

Notes:
- Creating a _new_ album from a selection of assets in the album view doesn't reload the `AssetGrid` / page correctly.
